File: Data Structures & Algorithms/valid-sudoku/submission-2.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        
        std::unordered_map<int, std::set<char>> rows;
        std::unordered_map<int, std::set<char>> cols;
        std::unordered_map<int, std::set<char>> sqrs;

        for (int row = 0; row < 9; row++){
            for (int col = 0; col < 9; col++){
                char c = board[row][col];
                if (c == '.') continue;

                if (
                    rows[row].count(c) || 
                    cols[col].count(c) || 
                    sqrs[(row / 3) * 3 + (col / 3)].count(c)
                    ) return false;
                
                rows[row].insert(c);
                cols[col].insert(c);
                sqrs[(row / 3) * 3 + (col / 3)].insert(c);

            }

        }
        return true;

    }
};

File: Data Structures & Algorithms/islands-and-treasure/submission-2.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        
        rows, cols = len(grid), len(grid[0])
        def dfs(row, col, count):

            if (row < 0 or row >= rows or
                col < 0 or col >= cols or
                grid[row][col] < count):
                return
            
            grid[row][col] = min(grid[row][col], count)
            dfs(row + 1, col, count + 1)
            dfs(row - 1, col, count + 1)
            dfs(row, col + 1, count + 1)
            dfs(row, col - 1, count + 1)
        
        for row in range(rows):
            for col in range(cols):
                if grid[row][col] == 0:
                    dfs(row, col, 0)
        
        


        